File: todo_mobile/lib/models/practice_question.dart
class PracticeQuestion {
  final String type; // "CHOICE", "CODE", "QA"
  final String title;
  final String description;
  final List<String>? options; // 仅选择题有
  final String? correctAnswer; // 仅选择题有
  final String? codeStub;      // 仅编程题有
  final String? explanation;   // 解析

  PracticeQuestion({
    required this.type,
    required this.title,
    required this.description,
    this.options,
    this.correctAnswer,
    this.codeStub,
    this.explanation,
  });

  factory PracticeQuestion.fromJson(Map<String, dynamic> json) {
    return PracticeQuestion(
      type: json['type'] ?? 'QA',
      title: json['title'] ?? '未知题目',
      description: json['description'] ?? '',
      options: json['options'] != null ? List<String>.from(json['options']) : [],
      correctAnswer: json['correctAnswer'],
      codeStub: json['codeStub'],
      explanation: json['explanation'],
    );
  }
}

File: todo_service/src/main/java/com/suat/app/backend/todo_service/dto/PracticeQuestion.java
package com.suat.app.backend.todo_service.dto;

import java.util.List;

public record PracticeQuestion(
        String type,          // "CHOICE", "CODE", "QA"
        String title,         // 题目标题
        String description,   // 详细描述
        List<String> options, // 选项 (仅用于 CHOICE, e.g., ["A. ...", "B. ..."])
        String correctAnswer, // 正确选项 (仅用于 CHOICE, e.g., "A")
        String codeStub,      // 代码模板 (仅用于 CODE)
        String explanation    // 解析/参考答案 (用于用户答题后查看)
) {}